block-producer/src/server/api.rs

```diff
@@ -30,6 +30,9 @@ impl<BB, TV> BlockProducerApi<BB, TV> {
 }
 
 #[tonic::async_trait]
+// FIXME: remove the allow when the upstream clippy issue is fixed:
+// https://github.com/rust-lang/rust-clippy/issues/12281
+#[allow(clippy::blocks_in_conditions)]
```

Comment:
is it possible to put it above the `async_trait` attribute like it's done for `state_view/mod.rs`?

Reply:
Done